services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -321,65 +327,423 @@ public void annotationsList() {
+
+  @Test
+  public void radiusesAreValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals(";unlimited;100", routeOptions.radiuses());
+  }
+
+  @Test
+  public void radiusesListIsValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals(3, routeOptions.radiusesList().size());
+    assertEquals(null, routeOptions.radiusesList().get(0));
+    assertEquals(Double.valueOf(Double.POSITIVE_INFINITY), routeOptions.radiusesList().get(1));
+    assertEquals(Double.valueOf(100.0), routeOptions.radiusesList().get(2));
+  }
+
+  @Test
+  public void bearingsAreValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals("0,90;90,0;", routeOptions.bearings());
+  }
+
+  @Test
+  public void bearingsListIsValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals(3, routeOptions.bearingsList().size());
+    assertEquals(Double.valueOf(0), routeOptions.bearingsList().get(0).get(0));
+    assertEquals(Double.valueOf(90), routeOptions.bearingsList().get(0).get(1));
+    assertEquals(Double.valueOf(90), routeOptions.bearingsList().get(1).get(0));
+    assertEquals(Double.valueOf(0), routeOptions.bearingsList().get(1).get(1));
+    assertEquals(null, routeOptions.bearingsList().get(2));
+  }
+
+  @Test
+  public void continueStraightIsValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals(false, routeOptions.continueStraight());
+  }
+
+  @Test
+  public void roundaboutExitsIsValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals(false, routeOptions.continueStraight());
+  }
+
+  @Test
+  public void geometriesAreValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals("polyline6", routeOptions.geometries());
+  }
+
+  @Test
+  public void stepsAreValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
 
-    assertEquals(ROUTE_OPTIONS_JSON, json);
+    assertEquals(true, routeOptions.steps());
   }
 
   @Test
-  public void waypointTargetsStringFromJson() {
+  public void annotationsAreValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals("congestion,distance,duration", routeOptions.annotations());
+  }
+
+  @Test
+  public void annotationsListIsValid_fromJson() {
+    RouteOptions routeOptions = RouteOptions.fromJson(ROUTE_OPTIONS_JSON);
+
+    assertEquals(3, routeOptions.annotationsList().size());
+    assertEquals("congestion", routeOptions.annotationsList().get(0));
+    assertEquals("distance", routeOptions.annotationsList().get(1));
+    assertEquals("duration", routeOptions.annotationsList().get(2));
+  }
+
+  @Test
+  public void excludeIsValid_fromJson() {
```

Comment:
Are we missing `toJson` test? Couldn't find it 👇

Reply:
there is a common `toJson` test to check all values
```
@Test
  public void routeOptions_toJson() {
    RouteOptions options = routeOptions();

    assertEquals(ROUTE_OPTIONS_JSON, options.toJson());
}
```
I added separate `toJson` tests only for values, which can be set either as a String or a list
for example `bearings(String)/bearingList(List)`

Reply:
> there is a common `toJson` test to check all values

👍 sounds good `assertEquals(ROUTE_OPTIONS_JSON, options.toJson());` we shouldn't rely on the order of keys but that's a different story. We should think about integrating https://github.com/JakeWharton/shimo/ in a future PR.